client/src/components/geneExpression/menus/createGenesetDialogue.js

```diff
@@ -0,0 +1,146 @@
+import _ from "lodash";
+import React from "react";
+import { connect } from "react-redux";
+import AnnoDialog from "../../categorical/annoDialog";
+import LabelInput from "../../categorical/labelInput";
+// import LabelInput from "../labelInput";
+// import { genesetPrompt, isGenesetErroneous } from "../genesetUtil";
+
+@connect((state) => ({
+  annotations: state.annotations,
+  schema: state.annoMatrix?.schema,
+  ontology: state.ontology,
+  obsCrossfilter: state.obsCrossfilter,
+  genesetsUI: state.genesetsUI,
+}))
+class CreateGenesetDialogue extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      genesetName: "",
+      genesToPopulateGeneset: "",
+    };
+  }
+
+  disableCreateGenesetMode = (e) => {
+    const { dispatch } = this.props;
+    this.setState({
+      genesetName: "",
+      genesToPopulateGeneset: "",
+    });
+    dispatch({
+      type: "geneset: disable create geneset mode",
+    });
+    if (e) e.preventDefault();
+  };
+
+  createGeneset = (e) => {
+    const { dispatch } = this.props;
+    const { genesetName, genesToPopulateGeneset } = this.state;
+
+    dispatch({
+      type: "geneset: create",
+      genesetName,
+    });
+    if (genesToPopulateGeneset) {
+      const genesTmpHardcodedFormat = [];
+
+      const genesArrayFromString = _.pull(
+        _.uniq(genesToPopulateGeneset.split(/[ ,]+/)),
+        ""
+      );
+
+      genesArrayFromString.forEach((_gene) => {
+        genesTmpHardcodedFormat.push({
+          geneSymbol: _gene,
+        });
+      });
+
+      dispatch({
+        type: "geneset: add genes",
+        genesetName,
+        genes: genesTmpHardcodedFormat,
+      });
+    }
+    dispatch({
+      type: "geneset: disable create geneset mode",
+    });
+    this.setState({
+      genesetName: "",
+      genesToPopulateGeneset: "",
+    });
+    e.preventDefault();
+  };
+
+  genesetNameError = () => {
+    /* todo genesets validation */
+    return false;
+  };
+
+  handleChange = (e) => {
+    this.setState({ genesetName: e });
+  };
+
+  handleGenesetInputChange = (e) => {
+    this.setState({ genesToPopulateGeneset: e });
+  };
+
+  instruction = () => {
+    return "New, unique geneset";
+    /* todo genesets */
```

Comment:
I think there's a bit of cleanup here along with line 109 below. The TODOs have similar content.